search/lib/api/wfs.js

```diff
@@ -130,22 +136,78 @@ const CONFORMANCE_RESPONSE = {
   conformsTo: [
     'http://www.opengis.net/spec/wfs-1/3.0/req/core',
     'http://www.opengis.net/spec/wfs-1/3.0/req/oas30',
-    'http://www.opengis.net/spec/wfs-1/3.0/req/html',
     'http://www.opengis.net/spec/wfs-1/3.0/req/geojson'
   ]
 };
 
+const { Catalog } = require('../stac/catalog');
+
+async function getCatalog (request, response) {
+  // create parameter dictionary from browse_path_template and provided values
+  const browseTemplate = process.env.BROWSE_PATH.split('/');
+  const params = request.params['0'].split('/');
+  logger.debug(`browseTemplate = ${inspect(browseTemplate)}`);
+  logger.debug(`params = ${inspect(params)}`);
+  logger.debug(params.map((val, idx) => [browseTemplate[idx], val]));
+  // defining fromEntries can be removed if Node updated to 12
+  Object.fromEntries = l => l.reduce((a, [k, v]) => ({ ...a, [k]: v }), {});
+  const browseParams = Object.fromEntries(
+    params.map((val, idx) => [browseTemplate[idx], val])
+  );
+  logger.debug(`browseParams = ${inspect(browseParams)}`);
+
+  const provider = request.params.providerId;
+  const collection = request.params.collectionId;
+
+  // create catalog
+  const date = request.params['0'].replace(/\//g, '-');
+  const cat = new Catalog();
+  cat.stac_version = settings.stac.version;
+  cat.id = `${collection}-${date}`;
+  cat.title = `${collection} ${date}`;
+  cat.description = `${provider} sub-catalog for ${date}`;
+
+  // get path from event
+  const event = request.apiGateway.event;
+  const path = event.path.replace(/^(\/stac)/, '');
+  // add links
+  cat.createRoot(generateAppUrl(event, ''));
+  const selfUrl = generateAppUrl(event, path);
+  cat.createSelf(selfUrl);
+  cat.createParent(selfUrl.slice(0, selfUrl.lastIndexOf('/')));
+
+  const granParams = {
+    collection_concept_id: collection,
+    provider
+  };
+  const { year, month, day } = browseParams;
+  const facets = await cmr.getGranuleTemporalFacets(granParams, year, month, day);
+  if (day) {
+    facets.itemids.forEach(id => cat.addItem(id, granParams.provider, granParams.collection_concept_id, id));
+  } else if (month) {
+    facets.days.forEach(d => cat.addChild(`${year}-${month}-${d} catalog`, `/${d}`));
+  } else if (year) {
+    facets.months.forEach(m => cat.addChild(`${year}-${m} catalog`, `/${m}`));
+  }
+
+  response.status(200).json(cat);
```

Comment:
The `status(200)` is redundant when using the `.json` response since it defaults to 200.